src/WebJobs.Extensions.DurableTask/Bindings/ActivityTriggerAttributeBindingProvider.cs

```diff
@@ -108,7 +108,19 @@ private class ActivityTriggerBinding : ITriggerBinding
 
             public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
             {
-                var activityContext = (DurableActivityContext)value;
+                if (!(value is DurableActivityContext activityContext))
+                {
+                    if (!(value is string serializedInput))
+                    {
+                        throw new InvalidOperationException($"Cannot execute an Activity Trigger without a {nameof(DurableActivityContext)} or a {nameof(String)} that represents the serialized input.");
+                    }
+
+                    // Durable functions expects input as a JArray with one element.
+                    serializedInput = $"[{serializedInput}]";
+
+                    activityContext = new DurableActivityContext($"azure-portal-execution-{Guid.NewGuid().ToString()}", serializedInput);

```

Comment:
I'm not so sure about using `azure-portal-` in the string here since this could theoretically apply to other scenarios as well. I do recommend we add a comment explaining that we added this logic primarily for Azure portal manual invocation, however.

Reply:
I was unsure too. I thought customers may want to be able to filter out portal requests, but as you said, I am not positive Azure Portal is the only scenario this lights up (it technically is any request to the `/admin/{functionName}` endpoint I believe).